src/extra_rules.jl

```diff
@@ -172,9 +172,14 @@ function ChainRules.rrule(::DiffractorRuleConfig, ::Type{SArray{S, T, N, L}}, x:
 end
 
 function ChainRules.frule((_, ∂x), ::Type{SArray{S, T, N, L}}, x::NTuple{L,T}) where {S, T, N, L}
-    SArray{S, T, N, L}(x), SArray{S, T, N, L}(∂x.backing)
+    #TODO: we really shouldn't actually see the isa(∂x, AbstractZero) case since the frule should be called then
+    Δx = isa(∂x, AbstractZero) ? ∂x : SArray{S, T, N, L}(ChainRulesCore.backing(∂x))
```

Comment:
this isn't hit anymore
```suggestion
    Δx = SArray{S, T, N, L}(ChainRulesCore.backing(∂x))
```